src/operator-typings.ts

```diff
@@ -0,0 +1,44 @@
+  <T2, TResult>(second: ObservableOrPromise<T2>, project: (v1: T, v2: T2) => TResult): Observable<TResult>;
+  <T2, TResult>(second: ObservableOrPromise<T2>, project: (v1: T, v2: T2) => TResult): Observable<TResult>;
+  <T2, TResult>(array: [ObservableOrPromise<T2>], project: (v1: T, v2: T2) => TResult): Observable<TResult>;
+  <T2, T3>(second: ObservableOrPromise<T2>, third: ObservableOrPromise<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ObservableOrPromise<T2>, third: ObservableOrPromise<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ArrayOrIterator<T2>, third: ObservableOrPromise<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ArrayOrIterator<T2>, third: ArrayOrIterator<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ArrayOrIterator<T2>, third: ArrayOrIterator<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ObservableOrPromise<T2>, third: ArrayOrIterator<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ObservableOrPromise<T2>, third: ArrayOrIterator<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ArrayOrIterator<T2>, third: ObservableOrPromise<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(array: [ObservableOrPromise<T2>, ObservableOrPromise<T3>]): Observable<[T, T2, T3]>;
+  <T2, T3, TResult>(second: ObservableOrPromise<T2>, third: ObservableOrPromise<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ObservableOrPromise<T2>, third: ObservableOrPromise<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ArrayOrIterator<T2>, third: ObservableOrPromise<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ArrayOrIterator<T2>, third: ArrayOrIterator<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ArrayOrIterator<T2>, third: ArrayOrIterator<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ObservableOrPromise<T2>, third: ArrayOrIterator<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ObservableOrPromise<T2>, third: ArrayOrIterator<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ArrayOrIterator<T2>, third: ObservableOrPromise<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(array: [ObservableOrPromise<T2>, ObservableOrPromise<T3>], project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <TResult>(array: ObservableOrPromise<any>[], project?: Function): Observable<TResult[]>;
+  (...observables: Array<ObservableOrPromise<T>>): Observable<T[]>;
+  (...observables: Array<ArrayOrIterator<T>>): Observable<T[]>;
+  <R>(...observables: Array<ObservableOrPromise<T> | ((...values: Array<T>) => R)>): Observable<R>;
+  <R>(...observables: Array<ArrayOrIterator<T> | ((...values: Array<T>) => R)>): Observable<R>;
```

Comment:
This seems similar to patterns of some C# interfaces, including some of Rx operators like `combinelatest`. basically this'll allows to have better typing inference experiences (such as intellisense?). Matter of if this is necessarily required for typescript land with RxJS, not sure. Bit lean on we might not need this.. especially thinking about effort to hand crafted operator signature would be increased in great amount, since lot of Rx operators has similar parameter patterns.

p.s : In case of Rx.Net, these type are generated via t4 (https://github.com/Reactive-Extensions/Rx.NET/blob/859e6159cb07be67fd36b18c2ae2b9a62979cb6d/Rx.NET/Source/System.Reactive.Linq/Reactive/Linq/Observable/CombineLatest.Generated.tt)